### PR TITLE
AM-3481 GroupManager for in-memory Groups serving

### PR DESCRIPTION
**Pull request by lgw-gravitee**

### CODE REVIEW

File: gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-common/src/main/java/io/gravitee/am/gateway/handler/common/spring/CommonConfiguration.java
Comment (leleueri) [inline gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-common/src/main/java/io/gravitee/am/gateway/handler/common/spring/CommonConfiguration.java:1]: **comment**: I will probably change the config key (sync.roles.enabled & sync.groups.enabled) later in the PR managed by AM-3476. but let's keep it like this for now

File: gravitee-am-service/src/main/java/io/gravitee/am/service/impl/user/BaseUserEnhancer.java
[the remark this answers is not in the document]
Comment (lgw-gravitee) [reply gravitee-am-service/src/main/java/io/gravitee/am/service/impl/user/BaseUserEnhancer.java:1]: DefaultUserEnhancer is for mAPI

File: gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-common/src/main/java/io/gravitee/am/gateway/handler/common/spring/CommonConfiguration.java
Comment (lgw-gravitee) [reply gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-common/src/main/java/io/gravitee/am/gateway/handler/common/spring/CommonConfiguration.java:1]: sure.

File: gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-common/src/main/java/io/gravitee/am/gateway/handler/common/group/GroupManagerFacade.java
[the remark this answers is not in the document]
Comment (lgw-gravitee) [reply gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-common/src/main/java/io/gravitee/am/gateway/handler/common/group/GroupManagerFacade.java:1]: Thanks to interface and two implementations (one in-memory, another direct-db-query) we can make application configurable without if-else clause inside Group/RoleManager.

[the remark this answers is not in the document]
Comment (lgw-gravitee) [reply gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-common/src/main/java/io/gravitee/am/gateway/handler/common/group/GroupManagerFacade.java:1]: we agreed to removed Facades interfaces

File: gravitee-am-gateway/gravitee-am-gateway-standalone/gravitee-am-gateway-standalone-distribution/src/main/resources/config/gravitee.yml [hunk 1]
```diff
@@ -477,7 +477,10 @@ liquibase:
...
 #      rounds: 10
+
 # synchronized roles by event manager to reduce db calls from gateway
 sync:
+  groups:
+    enabled: true
   roles:
-    enabled: false
\ No newline at end of file
+    enabled: true
```
Comment (leleueri) [inline gravitee-am-gateway/gravitee-am-gateway-standalone/gravitee-am-gateway-standalone-distribution/src/main/resources/config/gravitee.yml:486]: **comment**: don't think it should be enabled by default

**Review by leleueri — APPROVED** [review]
made a comment, can you change it please? no need to ask for approval after that

File: gravitee-am-gateway/gravitee-am-gateway-standalone/gravitee-am-gateway-standalone-distribution/src/main/resources/config/gravitee.yml (hunk 1 above)
Comment (lgw-gravitee) [reply gravitee-am-gateway/gravitee-am-gateway-standalone/gravitee-am-gateway-standalone-distribution/src/main/resources/config/gravitee.yml]: obvious :)

**Pull request merged**